docs/architecture/adr-069-gov-improvements.md

```diff
@@ -0,0 +1,193 @@
+This second iteration on gov unlocked many possibilities by letting governance proposals contain any number of proposals.
+The last addition of gov has been expedited proposals (proposals that have a shorter voting period and a higher quorum, approval threshold).
+
+The community requested ([1], [4]) two additional proposals for improving governance choices. Those proposals would be useful when having protocol decisions made on specific choices or simplifying regular proposals that do not require high community involvement.
+
+Additionally, the SDK should allow chains to customize the tallying method of proposals (if they want to count the votes in another way). Currently, the Cosmos SDK counts votes proportionally to the voting power/stake. However, custom tallying could allow counting votes with a quadratic function instead.
+
+## Decision
+
+`x/gov` will integrate these functions and extract helpers and interfaces for extending the `x/gov` module capabilities.
+
+### Proposals
+
+Currently, all proposals are [`v1.Proposal`][5]. Optimistic and multiple choice proposals require a different tally logic, but the rest of the proposal stays the same to not create other proposal types, `v1.Proposal` will have an extra field:
+
+```protobuf
+enum ProposalType {
+  NORMAL = 0;
+  MULTIPLE_CHOICE = 1;
+  OPTIMISTIC = 2;
+  EXPEDITED = 3;
+}
+```
+
+Note, that expedited becomes a proposal type itself instead of a boolean on the `v1.Proposal` struct.
+
+> An expedited proposal is by design a normal proposal with a quicker voting period and higher threshold. A expedited proposals if fail, gets converted to a normal proposal
+
+An expedited optimistic proposal and an expedited multiple choice proposal do not make sense based on the definition above and is a proposal type instead of a proposal characteristic.
+
+#### Optimistic Proposal
+
+An optimistic proposal is a proposal that automatically passes unless a threshold a NO votes is reached.
+
+Voter can only vote NO on the proposal. If the NO threshold is reached, the optimistic proposal is converted to a normal proposal.
+
+Two governance parameters will be in added [`v1.Params`][5] to support optimistic proposals:
+
+```protobuf
+// optimistic_authorized_addreses is an optional governance parameter that limits the authorized accounts than can submit optimisitc proposals
+repeated string optimistic_authorized_addreses = 17 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+
+// Optimistic rejected threshold defines at which percentage of NO votes, the optimistic proposal should fail and be converted to a normal proposal.
+string optimistic_rejected_threshold = 18 [(cosmos_proto.scalar) = "cosmos.Dec"];
+```
+
+#### Multiple Choice Proposal
+
+A multiple choice proposal is a proposal where the voting options can be defined by the proposer.
+
+The number of voting option will be limited to a maximum of 4. A new vote option `SPAM` will be added and distinguished of those voting options. 
```

Comment:
Why limit to 4? I agree with limiting the number of options, but why not 6 or 8?

Reply:
Oh, I think I got it, this is because you want to keep the existing tally function which only handles 4 options.

Reply:
That would indeed simplify everything if we keep it 4, but we can maybe consider increasing it, if it is really wanted.

Reply:
Not sure if it's really wanted, but for sure this is a smart way to bring the feature with minimal changes :+1:

Reply:
I think it really depends how this feature would be practically used.

Have you guys thought of some example proposals that would be suited to use the multiple choices and how they may fit within the 4 options limits? Maybe worth providing these examples in the ADR (just a bulleted list)?
 
Off the top of my head, I could see e.g. voting on electing group of peoples for some task. And the options might be different sets of candidates, in which case limiting it to 4 kinda makes sense since I assume we also want to not overwhelm the voters.

I also like the fact that we are reusing the existing VoteOption a lot. No point in adding complexity if not needed.

I am missing an aspect, how do you handle the case when the proposer wants to have less than 4 options, say 3?

Reply:
This feature will be used for signaling proposals (given that a multiple choice won't execute any message).
This leaves to many possibilities: choosing the canonical bridge for a chain, elections as you said, or surveys (hopefully not too much 😅)

Voting on multiple choice proposal will make an extra query to get the options. That way, we will be able to refuse non set options.

Reply:
> This feature will be used for signaling proposals (given that a multiple choice won't execute any message).

Yeah, I appreciated a lot that this is just for text proposals. 

> This leaves to many possibilities: choosing the canonical bridge for a chain, elections as you said, or surveys (hopefully not too much 😅),..

Right, and if the community is civil enough they should be able to boil down the final choices to go on-chain to a max of 4 using forum discussions. 

He knew it
<img src="https://github.com/cosmos/cosmos-sdk/assets/12249307/40d97568-d781-4914-b1cd-f59a4e5fdd09" width="200">

Jokes aside, I think it's better to implement the feature reusing as much of the existing as possible for now as you are planning. We don't need to overcomplicate things unless there's an explicit need, and until this feature is out and tested in the field we can't really know.

> Voting on multiple choice proposal will make an extra query to get the options. That way, we will be able to refuse non set options.

This makes sense, thanks.